deps/rabbitmq_shovel/src/rabbit_amqp10_shovel.erl

```diff
@@ -175,10 +176,14 @@ dest_endpoint(#{shovel_type := dynamic,
 
 -spec handle_source(Msg :: any(), state()) ->
     not_handled | state() | {stop, any()}.
-handle_source({amqp10_msg, _LinkRef, Msg}, State) ->
-    Tag = amqp10_msg:delivery_id(Msg),
-    Payload = amqp10_msg:body_bin(Msg),
-    rabbit_shovel_behaviour:forward(Tag, #{}, Payload, State);
+handle_source({amqp10_msg, _LinkRef, Amqp10Msg}, State) ->
+    Tag = amqp10_msg:delivery_id(Amqp10Msg),
+    Mc = mc:init(mc_amqp, Amqp10Msg, #{
+        % these are required values, however I don't think they make sense for AMQP 1.0
+        exchange => <<>>,
+        routing_keys => [<<>>]
+    }),
```

Comment:
`exchange` and `routing_keys` seem to be required by the `mc` implementation. If it is not set, then some crashes can be expected.